core/src/main/java/uk/gov/justice/services/core/interceptor/InterceptorObserver.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.services.core.interceptor;
+
+import uk.gov.justice.services.core.annotation.AnyLiteral;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+
+/**
+ * Observes for {@link AfterDeploymentValidation} and adds all {@link Interceptor} implementations
+ * to the {@link InterceptorCache}
+ */
+public class InterceptorObserver implements Extension {
```

Comment:
If we are having a specific observer for Interceptor.class would it be worth putting this behind a pattern that all of our observers use? - just a consideration if this is preferred standard way to do this.

Even potentially templating the class to be used based on the implementation?